topics/admin/tutorials/ansible-galaxy/tutorial.md

```diff
@@ -894,10 +894,10 @@ The configuration is quite simple thanks to the many sensible defaults that are
 >    `galaxy_layout`                 | `root-dir`                                                       | This enables the `galaxy_root` Galaxy deployment layout: all of the code, configuration, tools, and mutable-data (like caches, location files, etc.) folders will live by default beneath `galaxy_root`. User data is stored under `file_path`, a variable we will set later.
 >    `galaxy_root`                   | `/srv/galaxy`                                                    | This is the root of the Galaxy deployment.
 >    `galaxy_user`                   | `{name: "{{ galaxy_user_name }}", shell: /bin/bash}`             | The user that Galaxy will run as.
->    `galaxy_commit_id`              | `release_23.0`                                                   | The git reference to check out, which in this case is the branch for Galaxy Release 23.0
+>    `galaxy_commit_id`              | `release_23.1`                                                   | The git reference to check out, which in this case is the branch for Galaxy Release 23.1
```

Comment:
```suggestion
>    `galaxy_commit_id`              | `release_23.0`                                                   | The git reference to check out, which in this case is the branch for Galaxy Release 23.0
```

The entire GAT has been tested with Release 23.0 only. Our goal is to be sure these tutorials keep working for admins, and given that we skipped 22.05 due to infrastructure changes, I feel we need to be a bit more cautious here.

xref https://github.com/galaxyproject/training-material/issues/4322

Reply:
It does not work with 23.0, either. Same conda error.

Reply:
Well that's a massive issue then, could you share in what way they're not resolving? Was there a bugfix in 23.1 that we should backport?

Reply:
Regardless, reverted back to 23.0

Reply:
Can you please share the conda error? I think that's the single most important part of this error is what's going wrong with conda. (Were there discussions in matrix I missed or so?)

Reply:
> Well that's a large issue then, could you share in what way they're not resolving? Was there a bugfix in 23.1 that we should backport?

Sorry, I should have been less concise: The conda resolution fails for both 23.0 and 23.1 for me. And that is because of a conflict between conda versions and the latest python that it fetches. Both are defined by the tutorial, and both 23.0 and 23.1 can be gotten to work by specifying a more recent conda version as I did.

Reply:
> Can you please share the conda error? I think that's the single most important part of this error is what's going wrong with conda. (Were there discussions in matrix I missed or so?)

```
      - conda=4.12.0 -> python[version='>=3.10,<3.11.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0']
    Your python: python=3.11
```

PS: I see your messages with a lag.

Reply:
Ahhh it's a py3.11 issue. Ok, then the solution is maybe instead to pin the specific python version required? @natefoo's miniconda role enables that: https://github.com/galaxyproject/ansible-miniconda/blob/main/defaults/main.yml#L17

that said i'm surprised it fails, since the default galaxy conda environment is supposed to be created with ancient 3.6 https://github.com/galaxyproject/ansible-miniconda/blob/main/defaults/main.yml#L43

Reply:
Yes, it seems so. Though, what is the advantage of downgrading python rather than upgrading conda?

Reply:
So I can definitely agree with upgrading conda! I don't *want* to downgrade python, it's just an unexpected result given the error message.

From your error message, for conda 4.12.0,  it's erroring on Python 3.11. Based on the role Python 3.6 _should_ have been installed, but for some reason I imagine it's using system python? So this is I guess down to your specific operating system, that python 3.11 is installed by default which doesn't support conda 4. 

So, yes, we should update conda. Hopefully that fixes it all.